### PR TITLE
Support for Nullable

### PR DESCRIPTION
**Pull request by dbarwikowski**
# Info
This Pull Request is related to issue no. 27 (#27) 

# Changes
Support for Nullable:
- Ints (16, 32, 64)
- UInts (16, 32, 64)
- Bytes (signed and unsigned)
- Floating points (Single, Double, Decimal)

### CODE REVIEW

File: src/Valit/Rules/Extensions/ValitRuleFloatExtensions.cs [hunk 1]
```diff
@@ -4,64 +4,130 @@ namespace Valit
...
-        }          
+            return rule.Satisfies(p => p <= value);
+        }
 
-        public static IValitRule<TObject, float> IsLessThanOrEqualTo<TObject>(this IValitRule<TObject, float> rule, float value)  where TObject : class
+        public static IValitRule<TObject, float?> IsLessThanOrEqualTo<TObject>(this IValitRule<TObject, float?> rule, float value) where TObject : class
         {
             rule.ThrowIfNull(ValitExceptionMessages.NullRule);
-            return rule.Satisfies(p =>  p <= value);
+            return rule.Satisfies(p => p.HasValue && p <= value);
```
Comment (arekbal) [inline src/Valit/Rules/Extensions/ValitRuleFloatExtensions.cs]: handle NaN...

File: src/Valit/Rules/Extensions/ValitRuleFloatExtensions.cs [hunk 2]
```diff
@@ -4,64 +4,130 @@ namespace Valit
...
+            return rule.Satisfies(p => p.HasValue && p >= value);
+        }
 
-        public static IValitRule<TObject, float> IsGreaterThanOrEqualTo<TObject>(this IValitRule<TObject, float> rule, float value)  where TObject : class
+        public static IValitRule<TObject, float> IsLessThanOrEqualTo<TObject>(this IValitRule<TObject, float> rule, float value) where TObject : class
         {
             rule.ThrowIfNull(ValitExceptionMessages.NullRule);
-            return rule.Satisfies(p =>  p >= value);
-        }          
+            return rule.Satisfies(p => p <= value);
```
Comment (arekbal) [inline src/Valit/Rules/Extensions/ValitRuleFloatExtensions.cs]: handle nan

File: src/Valit/Rules/Extensions/ValitRuleFloatExtensions.cs [hunk 3]
```diff
@@ -4,64 +4,130 @@ namespace Valit
...
+        }
 
-        public static IValitRule<TObject, float> IsLessThan<TObject>(this IValitRule<TObject, float> rule, float value)  where TObject : class
+        public static IValitRule<TObject, float?> IsGreaterThanOrEqualTo<TObject>(this IValitRule<TObject, float?> rule, float value) where TObject : class
         {
-			rule.ThrowIfNull(ValitExceptionMessages.NullRule);    
-			return rule.Satisfies(p =>  !Single.IsNaN(p) && !Single.IsNaN(value) && value > p);     
-		}         
+            rule.ThrowIfNull(ValitExceptionMessages.NullRule);
+            return rule.Satisfies(p => p.HasValue && p >= value);
```
Comment (arekbal) [inline src/Valit/Rules/Extensions/ValitRuleFloatExtensions.cs]: handle nan

File: src/Valit/Rules/Extensions/ValitRuleFloatExtensions.cs [hunk 4]
```diff
@@ -4,64 +4,130 @@ namespace Valit
...
-			return rule.Satisfies(p =>  !Single.IsNaN(p) && p < 0f); 
-		}
+            rule.ThrowIfNull(ValitExceptionMessages.NullRule);
+            return rule.Satisfies(p => !Single.IsNaN(p) && p < 0f);
+        }
+
+        public static IValitRule<TObject, float?> IsNegative<TObject>(this IValitRule<TObject, float?> rule) where TObject : class
+        {
+            rule.ThrowIfNull(ValitExceptionMessages.NullRule);
+            return rule.Satisfies(p => p.HasValue && !Single.IsNaN(p.Value) && p < 0f);
```
Comment (arekbal) [inline src/Valit/Rules/Extensions/ValitRuleFloatExtensions.cs]: better explicit p.Value at the end as the conversion is calling it anyway.

**Review by arekbal — DISMISSED** [review]
point to consider:
https://referencesource.microsoft.com/#mscorlib/system/nullable.cs,109
nullable null is being lower value than nullable with value... these two are comparable somehow lol...

**Comment by dbarwikowski** [comment]
Yep, still need to unwrap nullable... But not today, plx

**Review by GooRiOn — APPROVED** [review]
Well done :D

**Pull request merged**